ako-gateway-api/nodes/dequeue_ingestion.go

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2023-2024 VMware, Inc.
+ * All Rights Reserved.
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*   http://www.apache.org/licenses/LICENSE-2.0
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package nodes
+
+import (
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+
+	akogatewayapilib "github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/ako-gateway-api/lib"
+	akogatewayapiobjects "github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/ako-gateway-api/objects"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/lib"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/nodes"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/objects"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pkg/utils"
+)
+
+func DequeueIngestion(key string, fullsync bool) {
+	utils.AviLog.Infof("key: %s, msg: starting graph Sync", key)
+	objType, namespace, name := lib.ExtractTypeNameNamespace(key)
+
+	schema, valid := ConfigDescriptor().GetByType(objType)
+	if !valid {
+		return
+	}
+
+	gatewayNsNameList, found := schema.GetGateways(namespace, name, key)
+	if !found {
+		//returning due to error, cannot delete or update
+		utils.AviLog.Errorf("key: %s, msg: got error while getting k8s object", key)
+		return
+	}
+
+	if objType == lib.Gateway {
+		handleGateway(namespace, name, fullsync, key)
+	}
+
+	routeTypeNsNameList, found := schema.GetRoutes(namespace, name, key)
+	if !found {
+		utils.AviLog.Errorf("key: %s, msg: got error while getting object", key, objType)
+		return
+	}
+
+	utils.AviLog.Debugf("key: %s, msg: processing gateways %v and routes %v", key, gatewayNsNameList, routeTypeNsNameList)
+	for _, gatewayNsName := range gatewayNsNameList {
+
+		parentNs, _, parentName := lib.ExtractTypeNameNamespace(gatewayNsName)
+		modelName := lib.GetModelName(lib.GetTenant(), akogatewayapilib.GetGatewayParentName(parentNs, parentName))
+
+		modelFound, modelIntf := objects.SharedAviGraphLister().Get(modelName)
+		if !modelFound || modelIntf == nil {
+			utils.AviLog.Errorf("key: %s, msg: no model found: %s", key, modelName)
+			continue
+		}
+		if objType == utils.Secret {
+			handleGateway(namespace, parentName, fullsync, lib.Gateway+"/"+gatewayNsName)
+			continue
```

Comment:
Where are we saving the model and publishing to rest layer in this case?

Reply:
That code is already present in handleGateway function
load-balancer-and-ingress-services-for-kubernetes/ako-gateway-api/nodes/dequeue_ingestion.go: line 154 to 157